### ISSUE TITLE
error with Emacs 24 - Osx

### ISSUE THREAD

**Issue by pietro909** [fred-o/jekyll-modes#2]
Hi there,
thank you for sharing your script.
I can't use it now because I'm getting this error:

```
Lisp error: (void-function markdown-mode)
  markdown-mode()
  pm-polymode([object pm-polymode-multi-auto "markdown-1" [object pm-polymode-multi-auto "markdown" unbound pm-host/markdown polymode-minor-mode " PM" (pm-exporter/pandoc) nil nil nil polymode-mode-map nil unbound nil nil nil (jekyll/yaml-frontmatter jekyll/liquid-expression jekyll/liquid-tag) jekyll/pygments nil] unbound jekyll-markdown-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound])
  apply(pm-polymode [object pm-polymode-multi-auto "markdown-1" [object pm-polymode-multi-auto "markdown" unbound pm-host/markdown polymode-minor-mode " PM" (pm-exporter/pandoc) nil nil nil polymode-mode-map nil unbound nil nil nil (jekyll/yaml-frontmatter jekyll/liquid-expression jekyll/liquid-tag) jekyll/pygments nil] unbound jekyll-markdown-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound])
  call-next-method()
  pm-polymode-multi([object pm-polymode-multi-auto "markdown-1" [object pm-polymode-multi-auto "markdown" unbound pm-host/markdown polymode-minor-mode " PM" (pm-exporter/pandoc) nil nil nil polymode-mode-map nil unbound nil nil nil (jekyll/yaml-frontmatter jekyll/liquid-expression jekyll/liquid-tag) jekyll/pygments nil] unbound jekyll-markdown-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound])
  apply(pm-polymode-multi [object pm-polymode-multi-auto "markdown-1" [object pm-polymode-multi-auto "markdown" unbound pm-host/markdown polymode-minor-mode " PM" (pm-exporter/pandoc) nil nil nil polymode-mode-map nil unbound nil nil nil (jekyll/yaml-frontmatter jekyll/liquid-expression jekyll/liquid-tag) jekyll/pygments nil] unbound jekyll-markdown-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound])
  eieio-generic-call-primary-only(pm-initialize ([object pm-polymode-multi-auto "markdown-1" [object pm-polymode-multi-auto "markdown" unbound pm-host/markdown polymode-minor-mode " PM" (pm-exporter/pandoc) nil nil nil polymode-mode-map nil unbound nil nil nil (jekyll/yaml-frontmatter jekyll/liquid-expression jekyll/liquid-tag) jekyll/pygments nil] unbound jekyll-markdown-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound]))
  pm-initialize([object pm-polymode-multi-auto "markdown-1" [object pm-polymode-multi-auto "markdown" unbound pm-host/markdown polymode-minor-mode " PM" (pm-exporter/pandoc) nil nil nil polymode-mode-map nil unbound nil nil nil (jekyll/yaml-frontmatter jekyll/liquid-expression jekyll/liquid-tag) jekyll/pygments nil] unbound jekyll-markdown-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound])
  jekyll-markdown-mode()
  set-auto-mode-0(jekyll-markdown-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer 2015-08-18-create-scene.markdown> "~/workspace/webgl-with-threejs/_posts/2015-08-18-create-scene.markdown" nil nil "~/workspace/webgl-with-threejs/_posts/2015-08-18-create-scene.markdown" (6845682 16777218))
  find-file-noselect("~/workspace/webgl-with-threejs/_posts/2015-08-18-create-scene.markdown" nil nil t)
  find-file("~/workspace/webgl-with-threejs/_posts/2015-08-18-create-scene.markdown" t)
  call-interactively(find-file nil nil)
  command-execute(find-file)
```

I've installed the package from MELPA.
While compiling I got some warning about deprecated functions.

thank you,
pietro

**Comment by Toxaris** [comment]
I experienced the same problem. To work around it, I installed  `markdown-mode` and `yaml-mode` from MELPA.

Background: See https://github.com/vspinu/polymode/issues/69. Either `polymode` should figure out how to depend on the modes referenced there, or packages like `jekyll-modes` should add the dependencies. Maybe a good first step would be for `jekyll-modes` to depend on `markdown-mode` and `yaml-mode`.
